index.bs

```diff
@@ -568,13 +578,23 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered sets=].
 
-1. For each |authenticator| currently available on this platform, if
-    <code>|options|.{{MakeCredentialOptions/attachment}}</code> is [=present|not present=] or its value matches
-    |authenticator|'s [=attachment modality=], [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|. 
-
 1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is 
     "{{NotFoundError}}", and terminate this algorithm.
 
+1. For each [=authenticator=] currently available on this platform, process each option in 
+    |options|.{{MakeCredentialOptions/authenticatorSelection}} by checking if the [=authenticator=] satisfies the 
+    requirement of the option. 
+        1. Check if |attachment| is [=present|not present=] or its value matches
+            |authenticator|'s attachment modality, [=continue=].
+        1. If |requireResidentKey| is set to |true|, check if the |authenticator| is capable of storing 
```

Comment:
s/storing/creating a/?

Also, I wonder if this filter should be happening on the CTAP side. Could you give me read access to the CTAP repository so I can check if the latest version has a good place to put this filter?